pkl-server/pkl-server.gradle.kts

```diff
@@ -24,15 +24,62 @@ tasks.test {
   exclude("**/NativeServerTest.*")
 }
 
-val nativeTest by tasks.registering(Test::class) {
-  dependsOn(":pkl-cli:assembleNative")
+private fun Test.configureNativeTest() {
   testClassesDirs = files(tasks.test.get().testClassesDirs)
   classpath = tasks.test.get().classpath
   include("**/NativeServerTest.*")
 }
 
+val testMacExecutableAarch64 by tasks.registering(Test::class) {
+  dependsOn(":pkl-cli:macExecutableAarch64")
+  configureNativeTest()
+}
+
+val testMacExecutableAmd64 by tasks.registering(Test::class) {
+  dependsOn(":pkl-cli:macExecutableAmd64")
+  configureNativeTest()
+}
+
+val testLinuxExecutableAmd64 by tasks.registering(Test::class) {
+  dependsOn(":pkl-cli:linuxExecutableAmd64")
+  configureNativeTest()
+}
+
+val testLinuxExecutableAarch64 by tasks.registering(Test::class) {
+  dependsOn(":pkl-cli:linuxExecutableAarch64")
+  configureNativeTest()
+}
+
+val testAlpineExecutableAmd64 by tasks.registering(Test::class) {
+  dependsOn(":pkl-cli:alpineExecutableAmd64")
+  configureNativeTest()
+}
+
+val testWindowsExecutableAmd64 by tasks.registering(Test::class) {
+  dependsOn(":pkl-cli:windowsExecutableAmd64")
+  configureNativeTest()
+}
+
 val testNative by tasks.existing
 testNative {
-  dependsOn(nativeTest)
+  when {
+    buildInfo.os.isMacOsX -> {
+      dependsOn(testMacExecutableAarch64)
+      if (buildInfo.arch == "aarch64") {
+        dependsOn(testMacExecutableAmd64)
```

Comment:
We depend on `amd64` if the arch is `aarch64`? I'm not following the logic here. Shouldn't this be done like linux with separate bodies for each architecture?

Reply:
Ah, this logic was flipped around. Will fix.

We cross-build when on macOS, thanks to Rosetta. We can't do this on Linux, though.